graphql_test.go

```diff
@@ -1135,3 +1207,76 @@ func TestTime(t *testing.T) {
 		},
 	})
 }
+
+func TestValueCoercion(t *testing.T) {
```

Comment:
Those tests look better, but why so complicated with the `Result` type and an array of it? Wouldn't top level fields suffice to test what we want to test?
```
type Query {
	coerceFloat(arg: Float): Float
	[...]
}
```